### PR TITLE
core: move thread flags from flags.h to thread.h

### LINKED ISSUE #4338: move thread flags from `flags.h` to `thread.h`
I think moving the flags and renaming them to something like `THREAD_CREATE_STACKTEST` and then dropping the `core/include/flags.h` file would make sense?!

### PR DESCRIPTION
**Pull request by haukepetersen**
fixes #4338

### CODE REVIEW

File: core/include/thread.h [hunk 1]
```diff
@@ -94,6 +94,32 @@
...
+ */
+/**
+ * @brief Set the new thread to sleeping
+ **/
+#define THREAD_CREATE_SLEEPING          (1)
+
+/**
+ * @brief Currently not implemented
+ */
+#define THREAD_AUTO_FREE                (2)
```
Comment (Kijewski) [inline core/include/thread.h]: Just drop it and forget about it, maybe?

Comment (haukepetersen) [inline core/include/thread.h]: Had the same in mind, but I think it makes sense to do this in a separate PR - so my plan was to merge this first and open a second one afterwards

Comment (Kijewski) [inline core/include/thread.h]: You're right. Should be a PR on its own.

**Comment by kaspar030** [comment]
looks good to me. 1st ACK.

**Comment by haukepetersen** [comment]
@OlegHahm, @thomaseichinger, @Kijewski: one of you up to giving a 2nd ack?

**Comment by OlegHahm** [comment]
needs a rebase

**Comment by kaspar030** [comment]
@haukepetersen *ping*

**Comment by haukepetersen** [comment]
rebased

**Comment by OlegHahm** [comment]
I don't really like the long macro names, but I agree that this change makes sense and the new names are actually more aligned with our naming convention. ACK.

**Comment by haukepetersen** [comment]
rebased and adapted the new `gnrc_tftp` example

**Comment by haukepetersen** [comment]
Travis is happy and we have 2 ACKs -> go

**Comment by OlegHahm** [comment]
Dman, should have thought about it. Now I have to rebase again.

**Comment by haukepetersen** [comment]
sorry :-)
